1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,5 +1,61 @@
-function setAlarm() {}
+function setAlarm() {
+  const input = document.getElementById("alarmSet").value;
+  const setBtn = document.getElementById("set");
+  const stopBtn = document.getElementById("stop");
+  const timer = document.getElementById("timeRemaining");
 
+  // Make Set Alarm Button working:
+  setBtn.addEventListener("click", () => {
+    // check if the input is a positive integer.
+    input <= 0 || input === ""
+      ? alert("please insert a positive integer")
+      : alert("The count down will start now");
+
+    // saving input value ,before clearing it's value.
+    let time = input;
+    // clearing the input value.
+    document.getElementById("alarmSet").value = "";
+
+    // making
+    let timerInterval = setInterval(CountDown, 1000);
+
+    function CountDown() {
+      let min;
+      let sec;
+      if (time >= 0) {
+        // calculate minutes and seconds with input value.
+        min = Math.floor(time / 60);
+        min < 10 ? (min = `0${min}`) : min;
```

Comment:
Its really uncommon to see code like this where you have an expression but you don't use the return value from the expression. Prefer
```
if (min < 10) min = `0${min}`;
```

Reply:
I agree , Tried  "min < 10 ? (min = `0${min}`) " and got error so thought it's mandatory to have a return value , didn't know if I put "if" before it & remove " ? & : "  I can have a short code like this and will solve the Error . Thanks